sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusClientFactory.cs

```diff
@@ -35,7 +35,7 @@ internal class ServiceBusClientFactory
             logForwarder.Start();
         }
 
-        internal ServiceBusClient CreateClientFromSetting(string connection)
+        public ServiceBusClient CreateClientFromSetting(string connection)
```

Comment:
Are you going to be calling this from ScaleController? Because it seems that ServiceBusScaleMonitor only calls CreateAdministrativeClient internally - I would have thought that ScaleController would only need the APIs provided by the monitor.

Reply:
+1 on wondering why this needs to be public.

Reply:
This was needed because the instantiation of a ServiceBusScaleMonitor requires a ServiceBusReceiver, which we created through a ServiceBusClient. See code pointer [here](https://github.com/chiangvincent/ScaleController/blob/user/vchiang/msiIntegration/ScaleController/Triggers/ServiceBusTrigger.cs#:~:text=public%20override%20IScaleMonitor%20GetScaleMonitor()). Not sure if this is the best way to do this -- open to suggestions here

Reply:
I'd rather reduce the number of dependencies that the ServiceBusScaleMonitor requires. It looks like it basically needs the IConfiguration instance and then it could handle everything else internally, i.e. creating the ClientFactory, Receiver, AdminClient. We may still need to support the old flow where the ClientFactory is injected in the extensions, but that could be left as an internal constructor. For the public constructor, we should try to reduce dependencies.

Reply:
Created new public constructor for ServiceBusScaleMonitor, and removed changes for ServiceBusClientFactory (based on offline conversation). Let me know if this looks good!